packages/scm/src/browser/scm-widget.tsx

```diff
@@ -454,8 +457,8 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
             onMouseEnter={this.showHover}
             onMouseLeave={this.hideHover}
             ref={this.detectHover}
-            onClick={this.selectChange}
-            onDoubleClick={this.open}>
+            onClick={(corePreferences && corePreferences['workbench.list.openMode'] === 'singleClick') ? this.open : this.selectChange}
+            onDoubleClick={this.open} >
```

Comment:
Should not we guard here as well?

Reply:
You mean as to use the same preference for `onDoubleClick`?

Reply:
I meant:
`(corePreferences && corePreferences['workbench.list.openMode'] === 'doubleClick') ? this.open : this.selectChange`

Reply:
I wasn't sure if the behavior for `onDoubleClick` was supposed to be updated as well.
I don't think `selectChange` would make sense for a double click.

Reply:
then it should be no-op if `openMode` is not `doubleClick`